command/expand.go

```diff
@@ -16,22 +16,28 @@ func expandSource(
 	client storage.Storage,
 	followSymlinks bool,
 	srcurl *url.URL,
+	isRaw bool,
```

Comment:
There is another way to implement this `--raw` flag: 
Instead of introducing a new boolean parameter to this function we can create a new function that just returns a channel with single object. `cp` and `rm` commands could check `--raw` flag and decide to call either `expandSource` or this new function (`rawSource` for example).

```go
func rawSource(srcurl *url.URL, followSymlinks bool) <-chan *storage.Object {
	ch := make(chan *storage.Object, 1)
	if storage.ShouldProcessUrl(srcurl, followSymlinks) {
		ch <- &storage.Object{URL: srcurl}
	}
	close(ch)
	return ch
}
```

This implementation has following advantages:
- `expandSource` function does what it's name says
- No need to update `expandSource` tests

One drawback that comes to my mind is `cp` and `rm` commands will have one additional check for `--raw` flag, but it would not be more complicated that updating `expandSource` function. 

What do you think?

Reply:
I think this approach is better than with parameter, because in the other [PR](https://github.com/peak/s5cmd/pull/324), I also had to add another parameter (`exclude`) to `expandSource`, and if I add parameters to `expandSource` like this, then it would be very hard to follow. I will change to new function as you suggested.